### PR TITLE
Repository migration

### PR DESCRIPTION
**Pull request by manoger**
__( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩__

# New things:
- New GitHub repository
- '.gitmodules'
- '.gitignore'
- 'gradlew'
- 'settings.gradle'
- 'gradle/wrapper/gradle-wrapper.properties'

# Changes:
- HiveHeader 'username' to 'user id'
- 'Common' dependency to 'Pandora'(github submodule)

# Removed:
- User entity
- User Repository

__( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩( ᐛ )و ٩__

### CODE REVIEW

File: src/main/java/hive/mugshot/controller/MugshotController.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      throw new NotAcceptedFileFormatException();
+    }
+    imageStorer.storeImageProfile(userId,insertedImage,imageName);
+  }
+
+  @GetMapping(produces = MediaType.IMAGE_JPEG_VALUE)
+  public ResponseEntity<Resource> searchProfileImage(
+      @RequestHeader(name = AUTHENTICATED_USER_ID) final String userId
+  ){
+    Resource file = imageStorer.loadImage(userId,imageName);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/controller/MugshotController.java]:
```suggestion
    final var file = imageStorer.loadImage(userId,imageName);
```

File: src/main/java/hive/mugshot/controller/UtilsController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  @Autowired
+  public UtilsController(final ImageStorer imageStorer){
+    this.imageStorer=imageStorer;
+  }
+
+  @ResponseStatus(code = HttpStatus.OK, reason = "Random image generated and successfully stored")
+  @PostMapping("/generateRandomImage")
+  public void generateRandomImage(@RequestHeader(name = AUTHENTICATED_USER_ID) final String userId) {
+    var generatedImage=ImageUtils.generateRandomImage();
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/controller/UtilsController.java]:
```suggestion
    final var generatedImage=ImageUtils.generateRandomImage();
```

File: src/main/java/hive/mugshot/controller/UtilsController.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    imageStorer.storeImageProfile(userId,generatedImage,imageName);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/controller/UtilsController.java:29]:
```suggestion
    imageStorer.storeImageProfile (
        userId,
        ImageUtils.generateRandomImage(),
        imageName
    );
```

File: src/main/java/hive/mugshot/storage/ImageStorer.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  @Value("${hive.mugshot.image-directory-path}")
+  private String rootDir;
+  @Value("${hive.mugshot.profile-image-dimension}")
+  private int imageSizeInPixels;
+
+  public void storeImageProfile
+      (final String userDirectoryName,final MultipartFile insertedImage, String imageStoredName){
+    createDirectoryIfNotExist(userDirectoryName);
+    try {
+      var buff = ImageUtils.resizeImageToSquare(ImageIO.read(insertedImage.getInputStream()),imageSizeInPixels);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageStorer.java]:
```suggestion
      final var buff = ImageUtils.resizeImageToSquare(ImageIO.read(insertedImage.getInputStream()),imageSizeInPixels);
```

File: src/main/java/hive/mugshot/storage/ImageStorer.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+@Service
+public class ImageStorer {
+  @Value("${hive.mugshot.image-directory-path}")
+  private String rootDir;
+  @Value("${hive.mugshot.profile-image-dimension}")
+  private int imageSizeInPixels;
+
+  public void storeImageProfile
+      (final String userDirectoryName,final MultipartFile insertedImage, String imageStoredName){
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageStorer.java]:
```suggestion
      (final String userDirectoryName,final MultipartFile insertedImage, final String imageStoredName){
```

File: src/main/java/hive/mugshot/storage/ImageStorer.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    } catch (IOException e) {
+      throw new ImageProfileIOException(e);
+    }
+  }
+
+  public void storeImageProfile
+      (final String userDirectoryName,final BufferedImage insertedImage,final String imageStoredName){
+    createDirectoryIfNotExist(userDirectoryName);
+    try {
+      var buff = ImageUtils.resizeImageToSquare(insertedImage,imageSizeInPixels);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageStorer.java]:
```suggestion
      final var buff = ImageUtils.resizeImageToSquare(insertedImage,imageSizeInPixels);
```

File: src/main/java/hive/mugshot/storage/ImageStorer.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      ImageIO.write(buff, "jpg", createFullPathToTheFile(userDirectoryName, imageStoredName).toFile());
+    } catch (IOException e) {
+      throw new ImageProfileIOException(e);
+    }
+  }
+  public Resource loadImage(final String userDirectoryName,final String imageName) {
+    try {
+      var file = createFullPathToTheFile(userDirectoryName,imageName);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageStorer.java]:
```suggestion
      final var file = createFullPathToTheFile(userDirectoryName,imageName);
```

File: src/main/java/hive/mugshot/storage/ImageStorer.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+      var resource = new UrlResource(file.toUri());
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageStorer.java]:
```suggestion
      final var resource = new UrlResource(file.toUri());
```

File: src/main/java/hive/mugshot/storage/ImageStorer.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+        throw new ImageNotFoundException();
+      }
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
+      throw new RuntimeException();
+    }
+  }
+
+  public void deleteImage(final String userDirectoryName,final String imageName) {
+    var parentDir = createFullPathToTheFile(userDirectoryName,imageName);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageStorer.java]:
```suggestion
    final var parentDir = createFullPathToTheFile(userDirectoryName,imageName);
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.regex.Pattern;
+
+public final class ImageUtils {
+  private static final String IMAGE_PATTERN = "(^.+\\.(gif|png|bmp|jpeg|jpg)$)";
+
+  private ImageUtils(){
+  }
+
+  public static boolean validateIfHasAnImageAsExtension(final String image){
+    var pattern = Pattern.compile(IMAGE_PATTERN);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var pattern = Pattern.compile(IMAGE_PATTERN);
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    var matcher = pattern.matcher(image);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
   final var matcher = pattern.matcher(image);
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    return matcher.matches();
+  }
+
+  public static BufferedImage resizeImageToSquare(final BufferedImage inputtedImage,final int imageSizeInPixels) {
+    // multi-pass bilinear div 2
+    var bufferedImageWithNewSize = new BufferedImage(imageSizeInPixels, imageSizeInPixels, BufferedImage.TYPE_INT_RGB);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var bufferedImageWithNewSize = new BufferedImage(imageSizeInPixels, imageSizeInPixels, BufferedImage.TYPE_INT_RGB);
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+    var reSizer = bufferedImageWithNewSize.createGraphics();
+    var resizingMode=
+        (inputtedImage.getHeight()<imageSizeInPixels)
+            ? RenderingHints.VALUE_INTERPOLATION_NEAREST_NEIGHBOR
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java:26]: don't use if ternary
```suggestion
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+  public static boolean validateIfHasAnImageAsExtension(final String image){
+    var pattern = Pattern.compile(IMAGE_PATTERN);
+    var matcher = pattern.matcher(image);
+    return matcher.matches();
+  }
+
+  public static BufferedImage resizeImageToSquare(final BufferedImage inputtedImage,final int imageSizeInPixels) {
+    // multi-pass bilinear div 2
+    var bufferedImageWithNewSize = new BufferedImage(imageSizeInPixels, imageSizeInPixels, BufferedImage.TYPE_INT_RGB);
+    var reSizer = bufferedImageWithNewSize.createGraphics();
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var reSizer = bufferedImageWithNewSize.createGraphics();
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+            ? RenderingHints.VALUE_INTERPOLATION_NEAREST_NEIGHBOR
+            : RenderingHints.VALUE_INTERPOLATION_BILINEAR;
+    reSizer.setRenderingHint(RenderingHints.KEY_INTERPOLATION, resizingMode);
+    reSizer.drawImage(inputtedImage, 0, 0, imageSizeInPixels, imageSizeInPixels, null);
+    reSizer.dispose();
+    return bufferedImageWithNewSize;
+  }
+
+  public static BufferedImage generateRandomImage(){
+    var yellow=0xF6BD60;
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var yellow=0xF6BD60;
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+    var orange=0xE9724C;
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var orange=0xE9724C;
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+    var gray = 0xE8E9EB;
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var gray = 0xE8E9EB;
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+    var blue = 0x5C9EAD;
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var blue = 0x5C9EAD;
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+    var black = 0x313638;
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var black = 0x313638;
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+    var colorsCombinations=new int[][]{
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var colorsCombinations=new int[][]{
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+        {yellow,black},
+        {orange,black},
+        {blue,black},
+        {blue,gray},
+        {orange,yellow}
+    };
+    var combinationIndex=new Random().nextInt(colorsCombinations.length);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var combinationIndex=new Random().nextInt(colorsCombinations.length);
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+    var img=new BufferedImage(9,9,BufferedImage.TYPE_INT_RGB);
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var img=new BufferedImage(9,9,BufferedImage.TYPE_INT_RGB);
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+    var maxH=img.getHeight();
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var maxH=img.getHeight();
```

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+    var maxV=img.getWidth();
```
Comment (SamuelLevi) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]:
```suggestion
    final var maxV=img.getWidth();
```

File: src/test/java/hive/mugshot/controller/MugshotControllerTest.java [hunk 25]
```diff
@@ -0,0 +1,142 @@
...
+  @Value("${hive.mugshot.image-directory-path}")
+  private String rootDir;
+  @Value("${hive.mugshot.profile-image-name}")
+  private String imageName;
+
+  @Mock
+  private ImageStorer imageStorer;
+
+  private Resource createImageForTest(final String directoryName) throws Exception{
+    var file=new File(directoryName + "/ProfileImage.jpg");
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/controller/MugshotControllerTest.java]:
```suggestion
    final var file=new File(directoryName + "/ProfileImage.jpg");
```

File: src/test/java/hive/mugshot/controller/MugshotControllerTest.java [hunk 26]
```diff
@@ -0,0 +1,142 @@
...
+  }
+
+  private void createDirectoryForTest(final String directoryName) throws Exception{
+    Files.createDirectories(Paths.get(directoryName));
+  }
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+    var mugshotController = new MugshotController(imageStorer);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/controller/MugshotControllerTest.java]:
```suggestion
    final var mugshotController = new MugshotController(imageStorer);
```

File: src/test/java/hive/mugshot/controller/MugshotControllerTest.java [hunk 27]
```diff
@@ -0,0 +1,142 @@
...
+    ReflectionTestUtils.setField(mugshotController, "imageName", imageName);
+    mockMvc = MockMvcBuilders.standaloneSetup(mugshotController).build();
+    validDirectoryName = (rootDir + "/" + userId + "/");
+  }
+
+  @Test
+  public void givenValidImage_WhenImageRetrieved_then200andJpegImageTypeIsReturned() throws Exception{
+    createDirectoryForTest(validDirectoryName);
+    var resourceImage=createImageForTest(validDirectoryName);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/controller/MugshotControllerTest.java]:
```suggestion
    final var resourceImage = createImageForTest(validDirectoryName);
```

File: src/test/java/hive/mugshot/controller/MugshotControllerTest.java [hunk 28]
```diff
@@ -0,0 +1,142 @@
...
+    mockMvc.perform(
+        get("/")
+        .header(AUTHENTICATED_USER_ID, userId))
+        .andExpect(status().isNotFound())
+        .andDo(print());
+  }
+
+  @Test
+  public void givenUnsupportedMediaType_WhenImageUploaded_then415isReturned()throws Exception{
+      BufferedImage originalImage = new BufferedImage(512, 512, BufferedImage.TYPE_INT_RGB);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/controller/MugshotControllerTest.java]:
```suggestion
      final var originalImage = new BufferedImage(512, 512, BufferedImage.TYPE_INT_RGB);
```

File: src/test/java/hive/mugshot/controller/MugshotControllerTest.java [hunk 29]
```diff
@@ -0,0 +1,142 @@
...
+      ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/controller/MugshotControllerTest.java]:
```suggestion
      final var byteArrayOutputStream = new ByteArrayOutputStream();
```

File: src/test/java/hive/mugshot/controller/MugshotControllerTest.java [hunk 30]
```diff
@@ -0,0 +1,142 @@
...
+      mockMvc.perform(multipart("/")
+          .file(multipartFile)
+          .header(AUTHENTICATED_USER_ID, userId))
+          .andExpect(status().isUnsupportedMediaType())
+          .andDo(print());
+  }
+
+  @Test
+  public void givenWrongRequestBodyKey_WhenImageUploaded_then400isReturned() throws Exception{
+      byte[] EmptyFile = new byte[0];
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/controller/MugshotControllerTest.java]:
```suggestion
```

File: src/test/java/hive/mugshot/controller/MugshotControllerTest.java [hunk 31]
```diff
@@ -0,0 +1,142 @@
...
+      multipartFile = new MockMultipartFile("wrongBodyKey", "ProfileImage.jpeg", MediaType.IMAGE_JPEG_VALUE, EmptyFile);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/controller/MugshotControllerTest.java]:
```suggestion
      multipartFile = new MockMultipartFile("wrongBodyKey", "ProfileImage.jpeg", MediaType.IMAGE_JPEG_VALUE, new byte[0]);
```

File: src/test/java/hive/mugshot/controller/UtilsControllerTest.java [hunk 32]
```diff
@@ -0,0 +1,57 @@
...
+  @Value("${hive.mugshot.profile-image-name}")
+  private String imageName;
+
+  @Mock
+  private ImageStorer imageStorer;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+    var utilsController = new UtilsController(imageStorer);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/controller/UtilsControllerTest.java]:
```suggestion
    final var utilsController = new UtilsController(imageStorer);
```

File: src/test/java/hive/mugshot/storage/ImageUtilsTest.java [hunk 33]
```diff
@@ -0,0 +1,45 @@
...
+  public void resizeImageToSquare_whenInitialImageIsNotNull_expectSquareImageInConfiguredSize(){
+    var initialImage=new BufferedImage(10,50,BufferedImage.TYPE_INT_RGB);
+    var resizedImage=ImageUtils.resizeImageToSquare(initialImage,imageSizeInPixels);
+    Assert.assertEquals(imageSizeInPixels,resizedImage.getWidth());
+    Assert.assertEquals(imageSizeInPixels,resizedImage.getHeight());
+  }
+
+  @Test
+  public void whenGenerateRandomImageAndResizeToSquare_expectSquareImageInConfiguredSize(){
+    var generatedImage=ImageUtils.generateRandomImage();
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageUtilsTest.java]:
```suggestion
    final var generatedImage=ImageUtils.generateRandomImage();
```

File: src/test/java/hive/mugshot/storage/ImageUtilsTest.java [hunk 34]
```diff
@@ -0,0 +1,45 @@
...
+    var imageResized=ImageUtils.resizeImageToSquare(generatedImage,imageSizeInPixels);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageUtilsTest.java]:
```suggestion
    final var imageResized=ImageUtils.resizeImageToSquare(generatedImage,imageSizeInPixels);
```

File: src/test/java/hive/mugshot/storage/ImageUtilsTest.java [hunk 35]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  @Test
+  public void validateIfHasAnImageAsExtension_whenParameterMatchPattern_expectTrueValue(){
+    Assert.assertTrue(ImageUtils.validateIfHasAnImageAsExtension("a.gif"));
+  }
+
+  @Test
+  public void resizeImageToSquare_whenInitialImageIsNotNull_expectSquareImageInConfiguredSize(){
+    var initialImage=new BufferedImage(10,50,BufferedImage.TYPE_INT_RGB);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageUtilsTest.java]:
```suggestion
    final var initialImage=new BufferedImage(10,50,BufferedImage.TYPE_INT_RGB);
```

File: src/test/java/hive/mugshot/storage/ImageUtilsTest.java [hunk 36]
```diff
@@ -0,0 +1,45 @@
...
+    var resizedImage=ImageUtils.resizeImageToSquare(initialImage,imageSizeInPixels);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageUtilsTest.java]:
```suggestion
    final var resizedImage=ImageUtils.resizeImageToSquare(initialImage,imageSizeInPixels);
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 37]
```diff
@@ -0,0 +1,111 @@
...
+    assertEquals(imageSizeInPixels,image.getHeight());
+    assertEquals(imageSizeInPixels,image.getWidth());
+  }
+  @Test(expected = ImageNotFoundException.class)
+  public void loadImage_WhenFileNotExists_expectImageNotFoundException() {
+    imageStorer.loadImage(userId,imageName);
+  }
+  @Test(expected = InvalidPathException.class)
+  public void loadImage_WhenUserDirectoryNameContainsInvalidCharacter_expectInvalidPathException() {
+    var invalidCharacters="></\\^\"*?".toCharArray();
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
    final var invalidCharacters="></\\^\"*?".toCharArray();
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 38]
```diff
@@ -0,0 +1,111 @@
...
+    var oneInvalidCharacter= RandomStringUtils.random(1,invalidCharacters);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
    final var oneInvalidCharacter= RandomStringUtils.random(1,invalidCharacters);
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 39]
```diff
@@ -0,0 +1,111 @@
...
+  public void loadImage_WhenMultipartIsNotNull_expectConfiguredImageName() throws IOException {
+    imageStorer.storeImageProfile(userId,multipartFile,imageName);
+    var resource=imageStorer.loadImage(userId,imageName);
+    assertEquals(imageName,resource.getFile().getName());
+  }
+
+  @Test
+  public void loadImage_WhenMultipartIsNotNull_expectConfiguredImageSize() throws IOException {
+    imageStorer.storeImageProfile(userId,multipartFile,imageName);
+    var resource=imageStorer.loadImage(userId,imageName);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
    final var resource=imageStorer.loadImage(userId,imageName);
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 40]
```diff
@@ -0,0 +1,111 @@
...
+    var image=ImageIO.read(resource.getFile());
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
    final var image=ImageIO.read(resource.getFile());
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 41]
```diff
@@ -0,0 +1,111 @@
...
+  public void deleteImage_WhenImageNameAndUserDirectoryIsNotNull_expectFileDoesNotExists(){
+    imageStorer.deleteImage(userId,imageName);
+    Path path=Paths.get(rootDir,userId,imageName);
+    assertFalse(Files.exists(path));
+  }
+
+  @Test
+  public void loadImage_WhenMultipartIsNotNull_expectConfiguredImageName() throws IOException {
+    imageStorer.storeImageProfile(userId,multipartFile,imageName);
+    var resource=imageStorer.loadImage(userId,imageName);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
    final var resource=imageStorer.loadImage(userId,imageName);
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 42]
```diff
@@ -0,0 +1,111 @@
...
+  public void uploadImage_WhenBufferedImageIsProvided_expectFileExists(){
+    imageStorer.storeImageProfile(userId,initialImage,imageName);
+    Path path=Paths.get(rootDir,userId,imageName);
+    assertTrue(Files.exists(path));
+  }
+
+  @Test
+  public void deleteImage_WhenImageNameAndUserDirectoryIsNotNull_expectFileDoesNotExists(){
+    imageStorer.deleteImage(userId,imageName);
+    Path path=Paths.get(rootDir,userId,imageName);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 43]
```diff
@@ -0,0 +1,111 @@
...
+    assertFalse(Files.exists(path));
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
    assertFalse(Files.exists(Paths.get(rootDir,userId,imageName)));
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 44]
```diff
@@ -0,0 +1,111 @@
...
+  public void uploadImage_WhenMultipartFileIsProvided_expectFileExists(){
+    imageStorer.storeImageProfile(userId,multipartFile,imageName);
+    Path path=Paths.get(rootDir,userId,imageName);
+    assertTrue(Files.exists(path));
+  }
+
+  @Test
+  public void uploadImage_WhenBufferedImageIsProvided_expectFileExists(){
+    imageStorer.storeImageProfile(userId,initialImage,imageName);
+    Path path=Paths.get(rootDir,userId,imageName);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 45]
```diff
@@ -0,0 +1,111 @@
...
+    assertTrue(Files.exists(path));
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
    assertTrue(Files.exists(Paths.get(rootDir,userId,imageName)));
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 46]
```diff
@@ -0,0 +1,111 @@
...
+        MediaType.IMAGE_JPEG_VALUE,
+        byteArrayOutputStream.toByteArray()
+    );
+  }
+
+  @Test
+  public void uploadImage_WhenMultipartFileIsProvided_expectFileExists(){
+    imageStorer.storeImageProfile(userId,multipartFile,imageName);
+    Path path=Paths.get(rootDir,userId,imageName);
+    assertTrue(Files.exists(path));
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
    assertTrue(Files.exists(Paths.get(rootDir,userId,imageName)));
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 47]
```diff
@@ -0,0 +1,111 @@
...
+        "WhateverName.gif",
+        MediaType.IMAGE_JPEG_VALUE,
+        byteArrayOutputStream.toByteArray()
+    );
+  }
+
+  @Test
+  public void uploadImage_WhenMultipartFileIsProvided_expectFileExists(){
+    imageStorer.storeImageProfile(userId,multipartFile,imageName);
+    Path path=Paths.get(rootDir,userId,imageName);
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
```

File: src/test/java/hive/mugshot/storage/ImageStorerTest.java [hunk 48]
```diff
@@ -0,0 +1,111 @@
...
+  @Value("${hive.mugshot.profile-image-name}")
+  private String imageName;
+  @Value("${hive.mugshot.profile-image-dimension}")
+  private int imageSizeInPixels;
+
+
+  @Before
+  public void setUp() throws IOException {
+    initialImage=new BufferedImage(512,512,BufferedImage.TYPE_INT_RGB);
+    var byteArrayOutputStream = new ByteArrayOutputStream();
```
Comment (SamuelLevi) [inline src/test/java/hive/mugshot/storage/ImageStorerTest.java]:
```suggestion
    final var byteArrayOutputStream = new ByteArrayOutputStream();
```

File: src/test/java/hive/mugshot/controller/MugshotControllerTest.java [hunk 49]
```diff
@@ -0,0 +1,141 @@
...
+    Files.createDirectories(Paths.get(directoryName));
+  }
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+    final var mugshotController = new MugshotController(imageStorer);
+    ReflectionTestUtils.setField(mugshotController, "imageName", imageName);
+    mockMvc = MockMvcBuilders.standaloneSetup(mugshotController).build();
+    validDirectoryName = (rootDir + "/" + userId + "/");
```
Comment (eiling) [inline src/test/java/hive/mugshot/controller/MugshotControllerTest.java]: Can be final

Comment (eiling) [reply src/test/java/hive/mugshot/controller/MugshotControllerTest.java]: Can also be Path
`Paths.get(rootDir, userId)`

File: src/main/resources/application.properties [hunk 50]
```diff
@@ -0,0 +1,21 @@
+spring.application.name=smugshot-service
+server.port=9500
+eureka.client.service-url.default-zone=http://localhost:8761/eureka
+# H2
```
Comment (eiling) [inline src/main/resources/application.properties]: Remove Spring Data and H2 properties

File: src/main/java/hive/mugshot/storage/ImageUtils.java [hunk 51]
```diff
@@ -0,0 +1,65 @@
...
+        {orange,black},
+        {blue,black},
+        {blue,gray},
+        {orange,yellow}
+    };
+    final var combinationIndex=new Random().nextInt(colorsCombinations.length);
+    final var img=new BufferedImage(9,9,BufferedImage.TYPE_INT_RGB);
+    final var maxH=img.getHeight();
+    final var maxV=img.getWidth();
+    for(int vertical=0;vertical<maxV;vertical++){
```
Comment (eiling) [inline src/main/java/hive/mugshot/storage/ImageUtils.java]: Fix spacing

File: src/main/java/hive/mugshot/storage/ImageUtils.java (hunk 13 above)
Comment (eiling) [reply src/main/java/hive/mugshot/storage/ImageUtils.java:26]: No problem with ternary operator here.
Spacing needs to be fixed.

File: src/main/java/hive/mugshot/controller/UtilsController.java (hunk 3 above)
Comment (eiling) [reply src/main/java/hive/mugshot/controller/UtilsController.java:29]: Matter of preference

File: src/main/java/hive/mugshot/controller/MugshotController.java [hunk 52]
```diff
@@ -0,0 +1,59 @@
...
+
+  @GetMapping(produces = MediaType.IMAGE_JPEG_VALUE)
+  public ResponseEntity<Resource> searchProfileImage(
+      @RequestHeader(name = AUTHENTICATED_USER_ID) final String userId
+  ){
+    final var file = imageStorer.loadImage(userId,imageName);
+    return ResponseEntity.ok().header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"" + file.getFilename() + "\"").body(file);
+  }
+
+  @ResponseStatus(code = HttpStatus.NO_CONTENT, reason = "Profile image successfully deleted")
```
Comment (eiling) [inline src/main/java/hive/mugshot/controller/MugshotController.java]: Return 200 here

File: src/main/java/hive/mugshot/exception/NotAcceptedFileFormatException.java [hunk 53]
```diff
@@ -0,0 +1,7 @@
+package hive.mugshot.exception;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.UNSUPPORTED_MEDIA_TYPE, reason = "Media type unsupported")
+public class NotAcceptedFileFormatException extends RuntimeException{
```
Comment (eiling) [inline src/main/java/hive/mugshot/exception/NotAcceptedFileFormatException.java]: NotAccepted -> Unsupported

File: src/main/java/hive/mugshot/storage/ImageStorer.java [hunk 54]
```diff
@@ -0,0 +1,86 @@
...
+    try {
+      final var file = createFullPathToTheFile(userDirectoryName,imageName);
+      final var resource = new UrlResource(file.toUri());
+      if (resource.exists() || resource.isReadable()) {
+        return resource;
+      }else{
+        throw new ImageNotFoundException();
+      }
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
```
Comment (eiling) [inline src/main/java/hive/mugshot/storage/ImageStorer.java]: Avoid printing the full stack trace

**Review by eiling — CHANGES_REQUESTED** [review]

File: src/main/java/hive/mugshot/MugshotApplication.java [hunk 55]
```diff
@@ -0,0 +1,15 @@
+package hive.mugshot;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.cloud.netflix.eureka.EnableEurekaClient;
+
+@SpringBootApplication
+@EnableEurekaClient
+@EntityScan( basePackages = {"hive.entity"} )
```
Comment (eiling) [inline src/main/java/hive/mugshot/MugshotApplication.java]: Remove Spring Data config

**Review by eiling — APPROVED** [review]

**Review by SamuelLevi — APPROVED** [review]

**Pull request merged**